File: Data Structures & Algorithms/two-integer-sum/submission-1.kt
class Solution {
    fun twoSum(nums: IntArray, target: Int): IntArray {
        val map = mutableMapOf<Int, Int>()
        nums.forEachIndexed { i, num ->
            if (map.contains(target - num)) {
                return intArrayOf(map[target - num]!!, i)
            }
            map[num] = i;
        }
        throw Exception("Impossible!")
    }
}

File: Data Structures & Algorithms/is-anagram/submission-1.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        ms, mt = {}, {}
        for c in s:
            if c in ms:
                ms[c] += 1
            else:
                ms[c] = 1
        for c in t:
            if c in mt:
                mt[c] += 1
            else:
                mt[c] = 1
        return ms == mt

        